src/core/font_renderer.js

```diff
@@ -744,27 +736,50 @@ function compileCharString(charStringCode, cmds, font, glyphId) {
   parse(charStringCode);
 }
 
-const NOOP = [];
+const NOOP = "";
```

Comment:
I just realized that changing this line also requires that https://github.com/mozilla/pdf.js/blob/3f1d07a35e8641f0b85b1262eabfb399253ff0b4/src/core/font_renderer.js#L788 is changed as follows to ensure that we don't unnecessarily try to re-compile a glyph that's known to be corrupt.
```javascript
if (fn === undefined) {
```